TLM/TLM/U/UBuilder.cs

```diff
@@ -0,0 +1,136 @@
+namespace TrafficManager.U {
+    using System;
+    using System.Collections.Generic;
+    using ColossalFramework.UI;
+    using JetBrains.Annotations;
+    using TrafficManager.State.Keybinds;
+    using TrafficManager.U.Autosize;
+    using TrafficManager.UI;
+    using TrafficManager.Util;
+    using UnityEngine;
+
+    /// <summary>
+    /// Create an UI builder to populate a panel with good things: buttons, sub-panels, create a
+    /// drag handle and other controls.
+    /// </summary>
+    /// <typeparam name="TControl">The UI Component and ISmartSizableControl we're placing into the UI.</typeparam>
+    public class UBuilder {
+        public AtlasBuilder AtlasBuilder;
+
+        public UBuilder([CanBeNull] AtlasBuilder atlasBuilder = null) {
+            AtlasBuilder = atlasBuilder;
+        }
+
+        public static UBuilder Create(string abAtlasName,
+                                       string abLoadingPath,
+                                       IntVector2 abSizeHint) {
+            var ab = new AtlasBuilder(
+                atlasName: abAtlasName,
+                loadingPath: abLoadingPath,
+                sizeHint: abSizeHint);
+            return new UBuilder(ab);
+        }
+
+        /// <summary>
+        /// Generic window creation.
+        /// Creates a window with generic padding (4px). The size will be "fit to children".
+        /// </summary>
+        /// <typeparam name="TWindow">The window root panel type.</typeparam>
+        /// <param name="setupFn">Function called on to perform window post-setup.</param>
+        /// <returns>The new window panel.</returns>
+        public TWindow CreateWindow<TWindow>()
+            where TWindow : UIComponent, U.Autosize.ISmartSizableControl {
+            var parent = UIView.GetAView();
+            var window = (TWindow)parent.AddUIComponent(typeof(TWindow));
+
+            window.ResizeFunction((UResizer r) => { r.FitToChildren(); });
+            window.SetPadding(UPadding.Const());
+
+            return window;
+        }
+
+        /// <summary>
+        /// Create a button as a child of current UIBuilder. A new UIBuilder is returned.
+        /// </summary>
+        /// <typeparam name="TButton">The type of UIButton and ISmartSizableControl.</typeparam>
+        public TButton Button<TButton, TParent>(TParent parent)
```

Comment:
TParent is redundant. just use UIComponent
unless if you plan to add more restriction in the future.
does the parent also needs to be ISmartSizableControl?

Reply:
please remove all redundant Generic arguments in the rest of the file too.

Reply:
Done. Nice catch, thank you